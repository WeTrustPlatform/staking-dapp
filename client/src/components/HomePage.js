import React from 'react';
import Navbar from './Navbar';
import MainSection from './MainSection';
import FAQSection from './FAQSection';
import Footer from './Footer';
import ActivitiesSection from './ActivitiesSection';
import TopSection from './TopSection';

class HomePage extends React.Component {
  render() {
    return (
      <div>
        <Navbar />
        <TopSection color="light" />
        <MainSection color="dark" />
        <ActivitiesSection color="light" />
        <FAQSection color="dark" />
        <Footer color="light" />
      </div>
    );
  }
}

export default HomePage;
