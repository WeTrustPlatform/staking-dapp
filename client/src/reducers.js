import {
  WEB3_NETWORK_ID,
  WEB3_AVAILABLE,
  WEB3_UNLOCK_ACCOUNT,
  WEB3_LOCK_ACCOUNT,
  WEB3_TRST_BALANCE,
  WEB3_ACCOUNT_ACTIVITIES,
  WEB3_OVERALL_STATS,
  WEB3_CONTRACTS,
  WEB3_USERS_STATS,
  WEB3_CAUSES_STATS,
  UNSTAKE_EXIT,
  UNSTAKE_WARNING,
  UNSTAKE_PENDING,
  UNSTAKE_PENDING_BACKGROUND,
  UNSTAKE_FAILURE,
  UNSTAKE_SUCCESS,
} from './actions';

const initialState = {
  networkId: 'unknown',
  account: null,
  web3: null,
  trstBalance: '0',
  accountActivities: [],
  overallStats: {
    currentStakes: '0',
    averageStakes: '0',
    averageStakeInUSD: '0',
    currentStakers: '0',
  },
  contracts: {},
  usersStats: {}, // {<user>: {yourStakes: []}}
  causesStats: {}, // {<stakingId>:{amount, stakers, rank, name, isOnSpring}}
  unstakeProcess: {}, // { step, activityId }
};

function reducers(state = initialState, action) {
  switch (action.type) {
    case WEB3_NETWORK_ID:
      return Object.assign({}, state, {
        networkId: String(action.networkId),
      });
    case WEB3_AVAILABLE:
      return Object.assign({}, state, {
        web3: action.web3,
      });
    case WEB3_UNLOCK_ACCOUNT:
      return Object.assign({}, state, {
        account: action.account,
      });
    case WEB3_LOCK_ACCOUNT:
      return Object.assign({}, state, {
        account: null,
      });
    case WEB3_TRST_BALANCE:
      return Object.assign({}, state, {
        trstBalance: action.trstBalance,
      });
    case WEB3_ACCOUNT_ACTIVITIES:
      return Object.assign({}, state, {
        accountActivities: action.accountActivities,
      });
    case WEB3_OVERALL_STATS:
      return Object.assign({}, state, {
        overallStats: action.overallStats,
      });
    case WEB3_CONTRACTS:
      return Object.assign({}, state, {
        contracts: action.contracts,
      });
    case WEB3_CAUSES_STATS:
      return Object.assign({}, state, {
        causesStats: action.causesStats,
      });
    case WEB3_USERS_STATS:
      return Object.assign({}, state, {
        usersStats: action.usersStats,
      });
    case UNSTAKE_EXIT:
      return Object.assign({}, state, {
        unstakeProcess: {},
      });
    case UNSTAKE_PENDING:
    case UNSTAKE_PENDING_BACKGROUND:
    case UNSTAKE_FAILURE:
    case UNSTAKE_SUCCESS:
    case UNSTAKE_WARNING:
      return Object.assign({}, state, {
        unstakeProcess: {
          step: action.type,
          activity: action.activity,
        },
      });
    default:
      return state;
  }
}

export default reducers;
