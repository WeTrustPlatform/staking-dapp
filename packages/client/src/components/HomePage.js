import React from 'react';
import Navbar from './Navbar';
import MainSection from './MainSection';
import FAQSection from './FAQSection';
import Footer from './Footer';
import YourStakesSection from './YourStakesSection';
import TopSection from './TopSection';
import RankingSection from './RankingSection';
import WhyStakingSection from './WhyStakingSection';

class HomePage extends React.Component {
  render() {
    return (
      <div>
        <Navbar />
        <TopSection color="light" />
        <WhyStakingSection color="dark" />
        <MainSection color="light" />
        <YourStakesSection color="dark" />
        <RankingSection color="light" />
        <FAQSection color="dark" />
        <Footer color="light" />
      </div>
    );
  }
}

export default HomePage;
