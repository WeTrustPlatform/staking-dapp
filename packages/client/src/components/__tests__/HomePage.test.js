import React from 'react';
import ReactDOM from 'react-dom';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import reducers from '../../reducers';
import HomePage from '../HomePage';

const store = createStore(reducers);
function App() {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <HomePage />
      </BrowserRouter>
    </Provider>
  );
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it.skip('tests snapshot', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
