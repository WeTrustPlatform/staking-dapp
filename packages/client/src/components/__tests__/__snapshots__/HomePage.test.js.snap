// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests snapshot 1`] = `
<div>
  <div>
    <header
      className="MuiPaper-root-385 MuiPaper-elevation4-391 MuiAppBar-root-376 MuiAppBar-positionRelative-381 MuiAppBar-colorPrimary-383"
    >
      <div
        className="Navbar-universalBar-374"
      >
        <a
          className="HrefLink-link-412"
          href="https://wetrust.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="WeTrust"
            className="Navbar-universalBarItem-375"
            src="we-trust-global-logo.svg"
          />
        </a>
        <a
          className="HrefLink-link-412"
          href="https://spring.wetrust.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Spring"
            className="Navbar-universalBarItem-375"
            src="spring-global-logo.svg"
          />
        </a>
        <a
          className="HrefLink-link-412"
          href="https://staking.wetrust.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="Staking"
            className="Navbar-universalBarItem-375"
            src="staking-global-logo.svg"
          />
        </a>
        <a
          className="HrefLink-link-412"
          href="https://tlc.wetrust.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="TLC"
            className="Navbar-universalBarItem-375"
            src="trusted-lending-circles-global-logo.svg"
          />
        </a>
        <a
          className="HrefLink-link-412"
          href="https://cryptounlocked.wetrust.io"
          rel="noopener noreferrer"
          target="_blank"
        >
          <img
            alt="CryptoUnlocked"
            className="Navbar-universalBarItem-375"
            src="crypto-unlocked-global-logo.svg"
          />
        </a>
      </div>
      <div
        className="MuiToolbar-root-413 MuiToolbar-regular-415 MuiToolbar-gutters-414"
      >
        <div
          className="Navbar-grow-372"
        >
          <img
            alt="WeTrust"
            className="Navbar-brandImage-373"
            src="wetrust-logo.svg"
          />
        </div>
        <ul
          className="MuiList-root-423 MuiList-padding-424"
        >
          <div
            className="Web3Account-accountContainer-421"
          >
            <div
              className="Web3Account-accountImageWrapper-422"
            >
              <img
                alt="metamask fox icon"
                height={21}
                src="meta-mask-logo.png"
                width={21}
              />
            </div>
            <div>
              <p
                className="MuiTypography-root-427 MuiTypography-body1-436 Web3Account-accountText-419"
              >
                Connect to Privatenet-5777
              </p>
            </div>
          </div>
        </ul>
      </div>
    </header>
  </div>
  <div
    className="Section-root-465 Section-light-467 TopSection-root-463"
    id="top-section"
  >
    <h4
      className="MuiTypography-root-427 MuiTypography-h4-442 MuiTypography-colorSecondary-458 MuiTypography-alignCenter-450 SectionHeader-container-468"
    >
      Help curate the SPRING directory
    </h4>
    <div
      className="MuiGrid-container-469 MuiGrid-justify-xs-center-486 TopSection-container-464"
    >
      <h6
        className="MuiTypography-root-427 MuiTypography-h6-444 MuiTypography-alignCenter-450"
      >
        As a decentralized fundraising platform, TRST token holders decide which causes should be highly ranked, and nominate new causes to join the platform
      </h6>
    </div>
  </div>
  <div
    className="Section-root-465 Section-dark-466"
  >
    <div
      className="MuiGrid-container-469 MuiGrid-justify-xs-center-486 WhyStakingSection-container-566"
    >
      <div
        className="MuiGrid-item-470 MuiGrid-grid-xs-12-509 MuiGrid-grid-sm-12-523 MuiGrid-grid-md-4-529 WhyStakingSection-gridItem-567"
      >
        <div>
          <img
            alt="increase-your-rank"
            src="icon-increase-rank.svg"
          />
        </div>
        <br />
        <h5
          className="MuiTypography-root-427 MuiTypography-h5-443"
        >
          Increase your Rank on the SPRING directory
        </h5>
        <br />
        <h6
          className="MuiTypography-root-427 MuiTypography-h6-444"
        >
          Stake your TRST tokens to increase visibility for your favorite Causes. Causes with the most stake assigned to them will receive higher ranking on SPRING.
        </h6>
      </div>
      <div
        className="MuiGrid-item-470 MuiGrid-grid-xs-12-509 MuiGrid-grid-sm-12-523 MuiGrid-grid-md-4-529 WhyStakingSection-gridItem-567 WhyStakingSection-midItem-568"
      >
        <div>
          <img
            alt="buy-trst"
            src="icon-buy-trst.svg"
          />
        </div>
        <br />
        <h5
          className="MuiTypography-root-427 MuiTypography-h5-443"
        >
          How to get Trustcoin
        </h5>
        <br />
        <h6
          className="MuiTypography-root-427 MuiTypography-h6-444"
        >
          You can obtain TRST tokens from the following platforms.
        </h6>
        <br />
        <div
          className="WhyStakingSection-row-569"
        >
          <div
            className="WhyStakingSection-rowItem-570"
          >
            <a
              href="https://www.bancor.network/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="bancor"
                height="54"
                src="bancor.png"
                width="38"
              />
            </a>
          </div>
          <div
            className="WhyStakingSection-rowItem-570"
          >
            <a
              href="https://uniswap.io/"
              rel="noopener noreferrer"
              target="_blank"
            >
              <img
                alt="uniswap"
                height="42"
                src="uniswap.png"
                width="98"
              />
            </a>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-470 MuiGrid-grid-xs-12-509 MuiGrid-grid-sm-12-523 MuiGrid-grid-md-4-529 WhyStakingSection-gridItem-567"
      >
        <div>
          <img
            alt="nominate"
            src="icon-nominate.svg"
          />
        </div>
        <br />
        <h5
          className="MuiTypography-root-427 MuiTypography-h5-443"
        >
          Nominate any organization
        </h5>
        <br />
        <h6
          className="MuiTypography-root-427 MuiTypography-h6-444"
        >
          Stake TRST tokens to nominate and reserve the top spot for your favorite Causes. Nominated Causes that have staked TRST will receive preferential placement upon joining the platform.
        </h6>
      </div>
    </div>
  </div>
  <div
    className="Section-root-465 Section-light-467"
    id="main-section"
  >
    <h4
      className="MuiTypography-root-427 MuiTypography-h4-442 MuiTypography-colorSecondary-458 MuiTypography-alignCenter-450 SectionHeader-container-468"
    >
      Start staking
    </h4>
    <div
      className="StakeNow-root-571"
    >
      <div
        aria-expanded={false}
        aria-haspopup="listbox"
        aria-labelledby="downshift-1-label"
        aria-owns={null}
        className="SearchInput-container-576"
        role="combobox"
      >
        <div
          className="MuiFormControl-root-578 MuiFormControl-marginNormal-579 MuiFormControl-fullWidth-581"
        >
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-1-label"
            className="MuiInputBase-root-595 MuiInput-root-582 MuiInputBase-fullWidth-604 MuiInput-fullWidth-589 MuiInputBase-formControl-596 MuiInput-formControl-583"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoComplete="off"
              className="MuiInputBase-input-605 MuiInput-input-590 MuiInputBase-inputType-608 MuiInput-inputType-593 MuiInputBase-inputTypeSearch-609 MuiInput-inputTypeSearch-594"
              disabled={false}
              id="downshift-1-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Search cause name"
              required={false}
              type="search"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-container-469 MuiGrid-justify-xs-center-486 StakeNow-stake-572"
      >
        <div
          className="MuiGrid-item-470 MuiGrid-grid-xs-10-507 MuiGrid-grid-sm-8-519 MuiGrid-grid-md-4-529 StakeNow-button-575"
        >
          <button
            className="MuiButtonBase-root-638 MuiButton-root-612 MuiButton-contained-623 MuiButton-containedPrimary-624 MuiButton-raised-626 MuiButton-raisedPrimary-627 MuiButton-sizeLarge-636 MuiButton-fullWidth-637"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex="0"
            type="button"
          >
            <span
              className="MuiButton-label-613"
            >
              Stake Now
            </span>
            <span
              className="MuiTouchRipple-root-736"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-container-469 MuiGrid-justify-xs-center-486"
    >
      <h6
        className="MuiTypography-root-427 MuiTypography-h6-444"
      >
        <a
          className="HrefLink-link-412"
          href="https://www.bancor.network/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Get more TRST
        </a>
      </h6>
    </div>
  </div>
  <div
    className="Section-root-465 Section-dark-466"
    id="activities-section"
  >
    <h4
      className="MuiTypography-root-427 MuiTypography-h4-442 MuiTypography-colorSecondary-458 MuiTypography-alignCenter-450 SectionHeader-container-468"
    >
      Your Stakes
    </h4>
    <div
      className="MuiPaper-root-385 MuiPaper-elevation2-389 MuiPaper-rounded-386 YourStakesSection-paper-661"
    >
      <table
        className="MuiTable-root-668 YourStakesSection-table-662"
      >
        <thead
          className="MuiTableHead-root-669"
        >
          <tr
            className="MuiTableRow-root-670 MuiTableRow-head-673"
          >
            <th
              aria-sort={null}
              className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingDense-680"
              scope="col"
            >
              Cause name
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingDense-680"
              scope="col"
            >
              Stake amount
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingDense-680"
              scope="col"
            >
              Locked until
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingDense-680"
              scope="col"
            >
              Status
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingDense-680"
              scope="col"
            >
              Tx hash
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root-687"
        >
          <tr
            className="MuiTableRow-root-670"
          >
            <td
              aria-sort={null}
              className="MuiTableCell-root-675 MuiTableCell-body-677 MuiTableCell-paddingDense-680"
              colSpan={5}
            >
              <div
                className="YourStakesSection-noActivities-666"
              >
                You have 0 transaction in Your Stakes history. 
                <a
                  href="/#main-section"
                  onClick={[Function]}
                >
                  Please stake!
                </a>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    className="Section-root-465 Section-light-467"
  >
    <div
      className="MuiGrid-container-469 MuiGrid-justify-xs-center-486 RankingSection-container-689"
    >
      <div
        className="MuiGrid-item-470 MuiGrid-grid-xs-12-509 MuiGrid-grid-sm-12-523 MuiGrid-grid-md-6-531 RankingSection-gridItem-690"
      >
        <div>
          <div>
            <h5
              className="MuiTypography-root-427 MuiTypography-h5-443"
            >
              Most staked Causes
            </h5>
          </div>
          <div
            className="LeaderBoard-subtitle-692"
          >
            <h6
              className="MuiTypography-root-427 MuiTypography-h6-444"
            >
              The following causes have been granted higher ranking by Trustcoin holders.
            </h6>
          </div>
          <div
            className="LeaderBoard-causesStats-693"
          >
            <div
              className="MuiPaper-root-385 MuiPaper-elevation2-389 MuiPaper-rounded-386"
            >
              <table
                className="MuiTable-root-668"
              >
                <thead
                  className="MuiTableHead-root-669"
                >
                  <tr
                    className="MuiTableRow-root-670 MuiTableRow-head-673"
                  >
                    <th
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingNone-682"
                      scope="col"
                    >
                      Rank
                    </th>
                    <th
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingNone-682"
                      scope="col"
                    >
                      Cause name
                    </th>
                    <th
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingNone-682"
                      scope="col"
                    >
                      Staked amount
                    </th>
                    <th
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingNone-682"
                      scope="col"
                    >
                      Stakers
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="MuiTableBody-root-687 LeaderBoard-tableBody-697"
                >
                  <tr
                    className="MuiTableRow-root-670"
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-body-677 MuiTableCell-paddingNone-682"
                      colSpan={4}
                    >
                      <div
                        className="LeaderBoard-loading-696"
                      >
                        <div
                          className="MuiCircularProgress-root-699 MuiCircularProgress-colorSecondary-703 MuiCircularProgress-indeterminate-701"
                          role="progressbar"
                          style={
                            Object {
                              "height": 40,
                              "width": 40,
                            }
                          }
                        >
                          <svg
                            className="MuiCircularProgress-svg-704"
                            viewBox="22 22 44 44"
                          >
                            <circle
                              className="MuiCircularProgress-circle-705 MuiCircularProgress-circleIndeterminate-707"
                              cx={44}
                              cy={44}
                              fill="none"
                              r={19}
                              strokeWidth={6}
                              style={Object {}}
                            />
                          </svg>
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-item-470 MuiGrid-grid-xs-12-509 MuiGrid-grid-sm-12-523 MuiGrid-grid-md-6-531 RankingSection-gridItem-690 RankingSection-bottomItem-691"
      >
        <div>
          <div>
            <h5
              className="MuiTypography-root-427 MuiTypography-h5-443"
            >
              Nominated Causes
            </h5>
          </div>
          <div
            className="LeaderBoard-subtitle-692"
          >
            <h6
              className="MuiTypography-root-427 MuiTypography-h6-444"
            >
              The following causes have been nominated by Trustcoin holders.
            </h6>
          </div>
          <div
            className="LeaderBoard-causesStats-693"
          >
            <div
              className="MuiPaper-root-385 MuiPaper-elevation2-389 MuiPaper-rounded-386"
            >
              <table
                className="MuiTable-root-668"
              >
                <thead
                  className="MuiTableHead-root-669"
                >
                  <tr
                    className="MuiTableRow-root-670 MuiTableRow-head-673"
                  >
                    <th
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingNone-682"
                      scope="col"
                    >
                      Rank
                    </th>
                    <th
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingNone-682"
                      scope="col"
                    >
                      Cause name
                    </th>
                    <th
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingNone-682"
                      scope="col"
                    >
                      Staked amount
                    </th>
                    <th
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-head-676 MuiTableCell-paddingNone-682"
                      scope="col"
                    >
                      Stakers
                    </th>
                  </tr>
                </thead>
                <tbody
                  className="MuiTableBody-root-687 LeaderBoard-tableBody-697"
                >
                  <tr
                    className="MuiTableRow-root-670"
                  >
                    <td
                      aria-sort={null}
                      className="MuiTableCell-root-675 MuiTableCell-body-677 MuiTableCell-paddingNone-682"
                      colSpan={4}
                    >
                      <div
                        className="LeaderBoard-loading-696"
                      >
                        <div
                          className="MuiCircularProgress-root-699 MuiCircularProgress-colorSecondary-703 MuiCircularProgress-indeterminate-701"
                          role="progressbar"
                          style={
                            Object {
                              "height": 40,
                              "width": 40,
                            }
                          }
                        >
                          <svg
                            className="MuiCircularProgress-svg-704"
                            viewBox="22 22 44 44"
                          >
                            <circle
                              className="MuiCircularProgress-circle-705 MuiCircularProgress-circleIndeterminate-707"
                              cx={44}
                              cy={44}
                              fill="none"
                              r={19}
                              strokeWidth={6}
                              style={Object {}}
                            />
                          </svg>
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="Section-root-465 Section-dark-466"
    id="faq-section"
  >
    <h4
      className="MuiTypography-root-427 MuiTypography-h4-442 MuiTypography-colorSecondary-458 MuiTypography-alignCenter-450 SectionHeader-container-468"
    >
      FAQ
    </h4>
    <div
      className="FAQSection-container-709"
    >
      <div
        className="FAQItem-root-710"
      >
        <div
          className="FAQItem-question-713"
          onClick={[Function]}
          role="presentation"
        >
          <div
            className="MuiGrid-container-469"
          >
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-10-507"
            >
              <h5
                className="MuiTypography-root-427 MuiTypography-h5-443"
              >
                General overview of Staking TRST on Spring
              </h5>
            </div>
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-2-499 FAQItem-gridIcon-712"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-715"
              >
                add
              </span>
            </div>
          </div>
        </div>
        <br />
        <hr
          className="MuiDivider-root-724 FAQItem-divider-711"
        />
      </div>
      <div
        className="FAQItem-root-710"
      >
        <div
          className="FAQItem-question-713"
          onClick={[Function]}
          role="presentation"
        >
          <div
            className="MuiGrid-container-469"
          >
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-10-507"
            >
              <h5
                className="MuiTypography-root-427 MuiTypography-h5-443"
              >
                What does staking TRST token enable?
              </h5>
            </div>
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-2-499 FAQItem-gridIcon-712"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-715"
              >
                add
              </span>
            </div>
          </div>
        </div>
        <br />
        <hr
          className="MuiDivider-root-724 FAQItem-divider-711"
        />
      </div>
      <div
        className="FAQItem-root-710"
      >
        <div
          className="FAQItem-question-713"
          onClick={[Function]}
          role="presentation"
        >
          <div
            className="MuiGrid-container-469"
          >
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-10-507"
            >
              <h5
                className="MuiTypography-root-427 MuiTypography-h5-443"
              >
                What is the benefit for staking TRST and why would I stake TRST?
              </h5>
            </div>
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-2-499 FAQItem-gridIcon-712"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-715"
              >
                add
              </span>
            </div>
          </div>
        </div>
        <br />
        <hr
          className="MuiDivider-root-724 FAQItem-divider-711"
        />
      </div>
      <div
        className="FAQItem-root-710"
      >
        <div
          className="FAQItem-question-713"
          onClick={[Function]}
          role="presentation"
        >
          <div
            className="MuiGrid-container-469"
          >
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-10-507"
            >
              <h5
                className="MuiTypography-root-427 MuiTypography-h5-443"
              >
                How does my stake benefit the organization or the crypto ecosystem?
              </h5>
            </div>
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-2-499 FAQItem-gridIcon-712"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-715"
              >
                add
              </span>
            </div>
          </div>
        </div>
        <br />
        <hr
          className="MuiDivider-root-724 FAQItem-divider-711"
        />
      </div>
      <div
        className="FAQItem-root-710"
      >
        <div
          className="FAQItem-question-713"
          onClick={[Function]}
          role="presentation"
        >
          <div
            className="MuiGrid-container-469"
          >
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-10-507"
            >
              <h5
                className="MuiTypography-root-427 MuiTypography-h5-443"
              >
                When can I unstake?
              </h5>
            </div>
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-2-499 FAQItem-gridIcon-712"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-715"
              >
                add
              </span>
            </div>
          </div>
        </div>
        <br />
        <hr
          className="MuiDivider-root-724 FAQItem-divider-711"
        />
      </div>
      <div
        className="FAQItem-root-710"
      >
        <div
          className="FAQItem-question-713"
          onClick={[Function]}
          role="presentation"
        >
          <div
            className="MuiGrid-container-469"
          >
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-10-507"
            >
              <h5
                className="MuiTypography-root-427 MuiTypography-h5-443"
              >
                What's next for TRST token usage after this?
              </h5>
            </div>
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-2-499 FAQItem-gridIcon-712"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-715"
              >
                add
              </span>
            </div>
          </div>
        </div>
        <br />
        <hr
          className="MuiDivider-root-724 FAQItem-divider-711"
        />
      </div>
      <div
        className="FAQItem-root-710"
      >
        <div
          className="FAQItem-question-713"
          onClick={[Function]}
          role="presentation"
        >
          <div
            className="MuiGrid-container-469"
          >
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-10-507"
            >
              <h5
                className="MuiTypography-root-427 MuiTypography-h5-443"
              >
                What are TRST tokens, where can I obtain them for staking purposes?
              </h5>
            </div>
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-2-499 FAQItem-gridIcon-712"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-715"
              >
                add
              </span>
            </div>
          </div>
        </div>
        <br />
        <hr
          className="MuiDivider-root-724 FAQItem-divider-711"
        />
      </div>
      <div
        className="FAQItem-root-710"
      >
        <div
          className="FAQItem-question-713"
          onClick={[Function]}
          role="presentation"
        >
          <div
            className="MuiGrid-container-469"
          >
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-10-507"
            >
              <h5
                className="MuiTypography-root-427 MuiTypography-h5-443"
              >
                How do I get my TRST tokens to show up in Metamask?
              </h5>
            </div>
            <div
              className="MuiGrid-item-470 MuiGrid-grid-xs-2-499 FAQItem-gridIcon-712"
            >
              <span
                aria-hidden="true"
                className="material-icons MuiIcon-root-715"
              >
                add
              </span>
            </div>
          </div>
        </div>
        <br />
        <hr
          className="MuiDivider-root-724 FAQItem-divider-711"
        />
      </div>
    </div>
  </div>
  <div
    className="Footer-container-731"
  >
    <div
      className="Footer-wrapper-732"
    >
      <a
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-link-730"
        href="/"
      >
        Home
      </a>
      <a
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-link-730"
        href="https://spring.wetrust.io/causes"
      >
        Causes
      </a>
      <a
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-link-730"
        href="https://www.wetrust.io"
      >
        About us
      </a>
      <a
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-link-730"
        href="https://github.com/WeTrustPlatform/documents/blob/master/WeTrust_Spring_Staking_ToS_v1.1.pdf"
      >
        Terms and Conditions
      </a>
      <a
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-link-730"
        href="https://github.com/WeTrustPlatform/documents/blob/master/FinclusionLabs_PrivacyPolicy_October92018_GDPRCompliant.pdf"
      >
        Privacy Policy
      </a>
    </div>
    <div
      className="Footer-wrapper-732"
    >
      <a
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-link-730 Footer-facebook-733"
        href="https://www.facebook.com/wetrustplatform"
      >
        Facebook
      </a>
      <a
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-link-730 Footer-twitter-734"
        href="https://twitter.com/wetrustplatform"
      >
        Twitter
      </a>
      <a
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-link-730 Footer-blog-735"
        href="https://blog.wetrust.io/"
      >
        Blog
      </a>
    </div>
    <div
      className="Footer-wrapper-732"
    >
      <p
        className="MuiTypography-root-427 MuiTypography-body1-436 Footer-copyright-729"
      >
        © 2019 WeTrustPlatform - All rights reserved
      </p>
    </div>
  </div>
</div>
`;
